src/amount.rs

```diff
@@ -266,10 +357,8 @@ impl Amount {
             } else {
                 (b.unit..a.unit).len() as u32
             };
-            let unit_base = *[a.unit, b.unit].iter().min().unwrap();
-
-            // println!("unit_distance_range: {:?}", (a.unit..b.unit));
-            // println!("unit_distance: {}", unit_distance);
+            // note: this unwrap_or() can never fail because there are two items.
+            let unit_base = *[a.unit, b.unit].iter().min().unwrap_or(&0);
```

Comment:
You can avoid all of the unwraps in this section by using [`std::cmp::min_by`](https://doc.rust-lang.org/std/cmp/fn.min_by.html):

```rust
let (base, other) = std::cmp::min_by(a, b, |a, b| a.unit.cmp(&b.unit));
let norm_base = NormalizedAmount {
    count: Integer::from(base.count),
    unit: base.unit
};

let norm_other = NormalizedAmount {
    count: Integer::from(10).pow(other.unit - base.unit),
    unit: base.unit
 }

(norm_base, norm_other)

Reply:
min_by is nice, but `let (base, other)` seems like black magic to me since min_by returns a single value T of same type as the input params.   anyway, this code has changed a bit and I came up with a condensed version.

Reply:
hehe, yes, idk what I was thinking